pkg/executor/docker/executor.go

```diff
@@ -200,39 +167,43 @@ func (e *Executor) Run(
 
 	if _, set := os.LookupEnv("SKIP_IMAGE_PULL"); !set {
 		dockerCreds := config.GetDockerCredentials()
-		if pullErr := e.client.PullImage(ctx, job.Spec.Docker.Image, dockerCreds); pullErr != nil {
-			pullErr = errors.Wrapf(pullErr, docker.ImagePullError, job.Spec.Docker.Image)
+		if pullErr := e.client.PullImage(ctx, dockerArgs.Image, dockerCreds); pullErr != nil {
+			pullErr = errors.Wrapf(pullErr, docker.ImagePullError, dockerArgs.Image)
 			return executor.FailResult(pullErr)
 		}
 	}
 
+	// TODO(forrest): [fixme] why are we doing this?
+	//    - was added via https://github.com/bacalhau-project/bacalhau/commit/7da22d9ab4c40ed721f832791b24abf73f14f308#diff-a5b36464e4d1c016e2a94b8fe22fdf2c58853f68d43b373ca42be9c16e2497a7R208-R217
 	// json the job spec and pass it into all containers
 	// TODO: check if this will overwrite a user supplied version of this value
 	// (which is what we actually want to happen)
-	log.Ctx(ctx).Debug().Msgf("Job Spec: %+v", job.Spec)
-	jsonJobSpec, err := model.JSONMarshalWithMax(job.Spec)
-	if err != nil {
-		return executor.FailResult(err)
-	}
-	log.Ctx(ctx).Debug().Msgf("Job Spec JSON: %s", jsonJobSpec)
-
-	useEnv := append(job.Spec.Docker.EnvironmentVariables,
-		fmt.Sprintf("BACALHAU_JOB_SPEC=%s", string(jsonJobSpec)),
-	)
+	/*
+			log.Ctx(ctx).Debug().Msgf("Job Spec: %+v", job.Spec)
+			jsonJobSpec, err := model.JSONMarshalWithMax(job.Spec)
+			if err != nil {
+				return executor.FailResult(err)
+			}
+			log.Ctx(ctx).Debug().Msgf("Job Spec JSON: %s", jsonJobSpec)
+
+		useEnv := append(job.Spec.Docker.EnvironmentVariables,
+			fmt.Sprintf("BACALHAU_JOB_SPEC=%s", string(jsonJobSpec)),
+		)
+	*/
```

Comment:
Need to address this before merging, I don't think it's necessary and I can't find any usage of the `BACALHAU_JOB_SPEC` environment variable. I believe there may have been some docker images that required access to the full job spec for some reason - I have linked to the commit that introduced this change in the code comment for the reviewer.